### ISSUE TITLE
Unable to select inner JSON fields with path in sensor (webhook gateway + JSON body)

### ISSUE THREAD

**Issue by brentnk** [argoproj/argo-events#441]
**Describe the bug**
To preface, I am not sure if the following behavior is intended or a bug.
If I follow the latest ["getting started"](https://github.com/argoproj/argo-events/blob/master/docs/getting_started.md) docs everything works as expected.  However, if I try to use the inner field as a param value:
```
        - src:
            event: "webhook-gateway:example"
            path: body.message
```
I get:
```
trigger failed to execute error="failed to apply params. err: unable to resolve 'webhook-gateway:example' parameter value. verify the path: 'body.message' is valid and/or set a default value for this param" trigger-n
ame=webhook-workflow-trigger
```

It looks like the value is being serialized as base64 because the [HTTP] body is being passed from the gateway as `byte[]`.  It looks like there was a similar issue with the calendar gateway at some point that was fixed #295 with the solution being to use `*json.RawMessage` in place of `[]byte`.

Just locally I was able to apply the same thing to the webhook gateway and that seemed to "fix" the issue.  Happy to provide more info and/or put in a PR for the fix.  Thanks!

**To Reproduce**
1.  Follow "getting started"
1. Add following to sensor
  ```
        - src:
            event: "webhook-gateway:example"
            path: body.message
  ```
1. Send an http message with the message field specified eg from docs `curl -d '{"message":"this is my first webhook"}' -H "Content-Type: application/json" -X POST $WEBHOOK_SERVICE_URL/example`
1. Check sensor pod logs for above error. 

**Expected behavior**
Expect to be able to select out inner fields of json.

**Environment (please complete the following information):**
- GKE: 1.14

**Comment by cristianoveiga** [comment]
Hi, I'm having the same issue.

Is the "path" key in the Trigger Parameterization now obsolete? How can I proceed instead?

**Comment by dark-solace** [comment]
Also having this issue

**Comment by theblackturtle** [comment]
Also having this issue

**Comment by VaibhavPage** [comment]
The `path` should be `message` instead of `body.message`.

**Comment by paveq** [comment]
In my test having path `message` did also not work, I just get error about `unable to resolve parameter value`

**Comment by cristianoveiga** [comment]
I've just tried using message @VaibhavPage and it didn't work. Same error message as @paveq .

**Comment by vakker** [comment]
I'm having the same issue, for me the logs from the pod (I changed the namespace to `argo-events`):
```
$ kubectl logs -n argo-events webhook-bld9q main
 _________________________________________ 
/ {"header":{"Accept":["*/*"],"Content-Le \
| ngth":["38"],"Content-Type":["applicati |
| on/json"],"User-Agent":["curl/7.58.0"]} |
| ,"body":"eyJtZXNzYWdlIjoidGhpcyBpcyBteS |
\ BmaXJzdCB3ZWJob29rIn0="}                /
 ----------------------------------------- 
    \
     \
      \     
                    ##        .            
              ## ## ##       ==            
           ## ## ## ##      ===            
       /""""""""""""""""___/ ===        
  ~~~ {~~ ~~~~ ~~~ ~~~~ ~~ ~ /  ===- ~~~   
       \______ o          __/            
        \    \        __/             
          \____\______/ 

$ argo get -n argo-events webhook-bld9q
Name:                webhook-bld9q
Namespace:           argo-events
ServiceAccount:      default
Status:              Succeeded
Created:             Thu Jan 09 16:12:59 +0000 (7 minutes ago)
Started:             Thu Jan 09 16:12:59 +0000 (7 minutes ago)
Finished:            Thu Jan 09 16:13:04 +0000 (7 minutes ago)
Duration:            5 seconds
Parameters:          
  message:           {"header":{"Accept":["*/*"],"Content-Length":["38"],"Content-Type":["application/json"],"User-Agent":["curl/7.58.0"]},"body":"eyJtZXNzYWdlIjoidGhpcyBpcyBteSBmaXJzdCB3ZWJob29rIn0="}

STEP              PODNAME        DURATION  MESSAGE
 ✔ webhook-bld9q  webhook-bld9q  5s 
```

It seems like the whole HTTP request is passed as an input and the body is encrypted maybe?

Could be relevant to #294.

**Comment by VaibhavPage** [comment]
v0.12 will fix this issue. The release will be out before next Friday.

**Comment by vakker** [comment]
@VaibhavPage  That's great, thanks!

In the meantime as a workaround, putting the data in the header works (even though it's a terrible workaround).
So change the original `resourceParameters` to:
```
      resourceParameters:
        - src:
            event: "webhook-gateway:example"
            path: header.Message.0
          dest: spec.arguments.parameters.0.value
```
And then:
```
$ curl -H "Content-Type: application/json" -H "message: asdasd" -X POST http://localhost:12000/example

$ kubectl logs -n argo-events webhook-4748d main 
 ________ 
< asdasd >
 -------- 
    \
     \
      \     
                    ##        .            
              ## ## ##       ==            
           ## ## ## ##      ===            
       /""""""""""""""""___/ ===        
  ~~~ {~~ ~~~~ ~~~ ~~~~ ~~ ~ /  ===- ~~~   
       \______ o          __/            
        \    \        __/             
          \____\______/
```

**Comment by VaibhavPage** [comment]
The fix is out in v0.12. Closing this issue for now.
